src/schemas/users.ts

```diff
@@ -0,0 +1,13 @@
+import { Type } from '@sinclair/typebox'
+
+const passwordPattern = '^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-]).*$'
+
+const PasswordSchema = Type.String({
+  pattern: passwordPattern,
```

Comment:
@climba03003 @jean-michelet 

Currently I don't think there is a way to customize the error message in case the payload doesn't match the pattern via the directly via the Type Builder Typebox.

I managed to do this via implementing fastify.setErrorHandler in the password update endpoint configuration, have you ever encountered this problem ?

Is it correct to manage the error message in a custom way at the endpoint ?

Reply:
> Is it correct to manage the error message in a custom way at the endpoint ?

You can using `ajv-errors`, so the message can be customize per schema.
https://www.npmjs.com/package/ajv-errors

Reply:
@jean-michelet 

Regarding the custom error message for AJV formatting, I haven't found a solution, and it seems that @climba03003  hasn't either ? as they approved the PR following my message.

https://github.com/fastify/demo/pull/48#discussion_r1835433228

As for SonarQube's warning about duplication, that's code I don't want to abstract and SonarQube doesn't know the difference

Reply:
We can review on the dashboard and claim it is fixed or safe.

Reply:
@Eomm 
Is it possible to give these rights to the Fastify members that collaborates on this repo? 
Besides me, I am aware of @climba03003 and @Fdawgs.

Reply:
I believe it can identify the Github account and give the correct permission?
Because I have updated the sonarcloud issue when I using the Github OAuth.